### ISSUE TITLE
Gloo Helm 3 support

### ISSUE THREAD

**Issue by GrahamGoudeau** [solo-io/gloo#1732]
Using this issue to track adding support for Helm 3

**Comment by GrahamGoudeau** [comment]
This should also include updating docs that reference Helm invocations, as some other things apart from the chart structure (`crds` directory https://github.com/helm/helm/pull/6243) have changed (ie, chart name is now a positional arg rather than given with `--name`)

**Comment by GrahamGoudeau** [comment]
Offline conversation with @kdorosh:

Going forward, our installation/uninstallation story is as follows:

- Perform installs/uninstalls entirely through Helm
- Carefully document the fact that gloo uninstalls will NOT clean up cluster-scoped resources (CRDs, RBAC) going forward, due to Helm design choices
- Document how to clean those resources up manually, if you would like a totally blank slate

**Comment by Vacant0mens** [comment]
Is there a test version out there that I could use to try out Gloo on Helm 3?

Using the current instructions in the [docs](https://docs.solo.io/gloo/latest/installation/gateway/kubernetes/) I'm getting:
```
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"  # x8
Error: failed pre-install: unable to build kubernetes object for deleting hook gloo/templates/18-settings.yaml: unable to recognize "": no matches for kind "Settings" in version "gloo.solo.io/v1"
```

I had to add `--generate-name` to get that, otherwise it would give me: `Error: must either provide a name or specify --generate-name`

**Comment by marcogschmidt** [comment]
@Vacant0mens Unfortunately, there currently is no version of the Gloo chart compatible with Helm 3. We are actively working in this and plan to release an updated chart by next week.

**Comment by Vacant0mens** [comment]
Looks like it's just waiting on some approvals now (or something. I'm not sure what your process is like).
@marcogschmidt is there something holding up the release?
Is there something I can do to use the new chart/repo today?

**Comment by kdorosh** [comment]
@Vacant0mens we released Gloo 1.2.0 yesterday, with helm 3 support. Thank you for your patience!

**Comment by Vacant0mens** [comment]
Having trouble uninstalling/re-installing the helm chart. I'm getting different errors like this:
```
# helm install gloo-gateway gloo/gloo --namespace gloo
Error: rendered manifests contain a resource that already exists. Unable to continue with install: existing resource conflict: kind: ClusterRoleBinding, namespace: , name: gateway-resource-reader-binding-gloo
```
I got similar errors about ClusterRole's, but then I removed all gloo-related ClusterRole's and ClusterRoleBinding's using `kubectl get ClusterRole | grep gloo` and `kubectl get ClusterRoleBinding | grep gloo` and then copy-paste-ed the two into `kubectl delete clusterrole` and `kubectl delete clusterrolebinding`.

Is this covered in [this](https://github.com/solo-io/gloo/issues/1938) issue? or should it be a new one?

**Comment by Vacant0mens** [comment]
@kdorosh 
also, my `gateway` pod isn't coming up all the way. The other three pods (`discovery`, `gloo`, and `gateway-proxy`) come up fine. The container is stuck in a `NotReady` state. Logs show:
```
{"level":"info","ts":1576266338.7889414,"logger":"gateway.v1.event_loop","caller":"v1/setup_event_loop.sk.go:57","msg":"event loop started","version":"1.2.8"}
{"level":"info","ts":1576266338.9253395,"logger":"gateway.v1.event_loop.gateway","caller":"validation/robust_client.go:34","msg":"starting proxy validation client... this may take a moment","version":"1.2.8","validation_server":"gloo:9988"}
```
But nothing else.

**Comment by kdorosh** [comment]
@Vacant0mens can you provide steps to reproduce these issues? (also which Gloo version are you running?)

I haven't been able to reproduce either issue on the newest release of Gloo.

We should definitely track both potential issues in separate, new issues.

**Comment by Vacant0mens** [comment]
Oh, no. Sorry @kdorosh , that was my bad. I was removing the Pods and Namespace specifically instead of running the `helm uninstall` command. That part works fine.

But the `gateway` pod/container still isn't starting all the way.

**Comment by Vacant0mens** [comment]
opened new issue [here](https://github.com/solo-io/gloo/issues/1964)
